command/jsonprovider/block.go

```diff
@@ -0,0 +1,69 @@
+package jsonprovider
+
+import (
+	"github.com/hashicorp/terraform/configs/configschema"
+)
+
+type block struct {
+	Attributes map[string]*attribute `json:"attributes,omitempty"`
+	BlockTypes map[string]*blockType `json:"block_types,omitempty"`
+}
+
+type blockType struct {
+	NestingMode string `json:"nesting_mode,omitempty"`
+	Block       *block `json:"block,omitempty"`
+	MinItems    uint64 `json:"min_items"`
+	MaxItems    uint64 `json:"max_items"`
+}
+
+func marshalBlockTypes(nestedBlock *configschema.NestedBlock) *blockType {
+	if nestedBlock == nil {
+		return &blockType{}
+	}
+	ret := &blockType{
+		Block:    marshalBlock(&nestedBlock.Block),
+		MinItems: uint64(nestedBlock.MinItems),
+		MaxItems: uint64(nestedBlock.MaxItems),
+	}
+
+	switch nestedBlock.Nesting.String() {
+	case "nestingModeInvalid":
+		ret.NestingMode = "invalid"
+	case "NestingSingle":
+		ret.NestingMode = "single"
+	case "NestingList":
+		ret.NestingMode = "list"
+	case "NestingSet":
+		ret.NestingMode = "set"
+	case "NestingMap":
+		ret.NestingMode = "map"
+	default:
+		// unpossible.
```

Comment:
Maybe move the statement from `"nestingModeInvalid"` down here and remove that case? I think it being literally `nestingModeInvalid` and it being some value that isn't in the enum at all are... both invalid, really.